src/sagas/task.saga.test.js

```diff
@@ -0,0 +1,382 @@
+import { call, put, select } from 'redux-saga/effects';
+import { cloneableGenerator } from 'redux-saga/utils';
+import * as childProcess from 'child_process';
+import * as path from 'path';
+import chalkRaw from 'chalk';
+
+import {
+  launchDevServer,
+  taskRun,
+  taskAbort,
+  displayTaskComplete,
+  taskComplete,
+  getBaseProjectEnvironment,
+  getDevServerCommand,
+} from './task.saga';
+import {
+  attachTaskMetadata,
+  receiveDataFromTaskExecution,
+  completeTask,
+  loadDependencyInfoFromDisk,
+} from '../actions';
+import killProcessId from '../services/kill-process-id.service';
+import { getProjectById } from '../reducers/projects.reducer';
+import { getPathForProjectId } from '../reducers/paths.reducer';
+import findAvailablePort from '../services/find-available-port.service';
+import { PACKAGE_MANAGER_CMD } from '../services/platform.service';
+import { ipcRenderer } from 'electron';
+
+const chalk = new chalkRaw.constructor({ level: 3 });
+
+// we want to watch and ensure that caught errors
+// are logged to the error log
+global.console = { error: jest.fn() };
```

Comment:
I noticed when running the tests that `console.log` fails ("console.log is not a function").

It also kinda worries me to do this, since it's never getting unset; I can imagine it causing problems for other tests run in the same session.

Maybe we can do this in a `beforeAll` block, and then unset it in an `afterAll`?

```js
// At the top, here:
const consoleOriginal = global.console;

// Within a test block:
beforeAll(() => {
  global.console = {
    ...consoleOriginal, // I wonder if this would work, to preserve `console.log`?
    error: jest.fn()
  };
});

afterAll(() => {
  global.console = consoleOriginal;
})
```

Reply:
It's only used in this single test, so I just moved it all to that block. There's also no need to replace `global.console` entirely when we can just replace `global.console.error`. Good catch.